File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/importering/autoimport/SykefraværsstatistikkImporteringUtils.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.importering.autoimport;

import java.math.BigDecimal;
import java.util.Arrays;
import java.util.List;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles.ÅrstallOgKvartal;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.importering.SykefraværsstatistikkVirksomhet;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.importering.SykefraværsstatistikkVirksomhetMedGradering;

public class SykefraværsstatistikkImporteringUtils {

  public static List<SykefraværsstatistikkVirksomhet> genererSykefraværsstatistikkVirksomhet(
      ÅrstallOgKvartal sisteKvartal) {
    return Arrays.asList(
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562452", "A", "2", 0,
            BigDecimal.valueOf(23.333500), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562452", "B", "2", 0,
            BigDecimal.valueOf(28.000000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562452", "C", "2", 0,
            BigDecimal.valueOf(58.920000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562452", "D", "2", 0,
            BigDecimal.valueOf(147.033898), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562452", "X", "2", 45,
            BigDecimal.valueOf(0.000000), BigDecimal.valueOf(2149.569700)),

        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311874411", "A", "2", 0,
            BigDecimal.valueOf(1.440000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311874411", "B", "2", 0,
            BigDecimal.valueOf(15.010000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311874411", "C", "2", 0,
            BigDecimal.valueOf(59.625000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311874411", "D", "2", 0,
            BigDecimal.valueOf(126.919576), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311874411", "E", "2", 0,
            BigDecimal.valueOf(75.614300), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311874411", "F", "2", 0,
            BigDecimal.valueOf(1.200000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311874411", "X", "2", 59,
            BigDecimal.valueOf(0.000000), BigDecimal.valueOf(2049.160900)),

        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "315829062", "A", "2", 0,
            BigDecimal.valueOf(9.133400), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "315829062", "C", "2", 0,
            BigDecimal.valueOf(20.969600), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "315829062", "D", "2", 0,
            BigDecimal.valueOf(39.537600), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "315829062", "X", "2", 49,
            BigDecimal.valueOf(0.000000), BigDecimal.valueOf(1333.733200)),

        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311545795", "A", "2", 0,
            BigDecimal.valueOf(12.600000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311545795", "B", "2", 0,
            BigDecimal.valueOf(14.000000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311545795", "C", "2", 0,
            BigDecimal.valueOf(4.600000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311545795", "D", "2", 0,
            BigDecimal.valueOf(75.018182), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311545795", "F", "2", 0,
            BigDecimal.valueOf(14.800000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "311545795", "X", "2", 43,
            BigDecimal.valueOf(0.000000), BigDecimal.valueOf(1831.816000)),

        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "312679671", "A", "2", 0,
            BigDecimal.valueOf(5.800000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "312679671", "B", "2", 0,
            BigDecimal.valueOf(22.500000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "312679671", "C", "2", 0,
            BigDecimal.valueOf(17.400000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "312679671", "D", "2", 0,
            BigDecimal.valueOf(1.000000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "312679671", "F", "2", 0,
            BigDecimal.valueOf(112.100000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "312679671", "X", "2", 59,
            BigDecimal.valueOf(0.000000), BigDecimal.valueOf(3208.506000)),

        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562436", "A", "2", 0,
            BigDecimal.valueOf(16.000000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562436", "B", "2", 0,
            BigDecimal.valueOf(61.000000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562436", "C", "2", 0,
            BigDecimal.valueOf(67.700000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562436", "D", "2", 0,
            BigDecimal.valueOf(355.151638), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562436", "E", "2", 0,
            BigDecimal.valueOf(250.400000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562436", "F", "2", 0,
            BigDecimal.valueOf(59.000000), BigDecimal.valueOf(0.000000)),
        new SykefraværsstatistikkVirksomhet(sisteKvartal.getÅrstall(), sisteKvartal.getKvartal(),
            "910562436", "X", "2", 469,
            BigDecimal.valueOf(0.000000), BigDecimal.valueOf(26694.646600))
        );
  }

  public static List<SykefraværsstatistikkVirksomhetMedGradering> genererSykefraværsstatistikkVirksomhetMedGradering(
      ÅrstallOgKvartal sisteKvartal) {
    return Arrays.asList(
        new SykefraværsstatistikkVirksomhetMedGradering(sisteKvartal.getÅrstall(),
            sisteKvartal.getKvartal(), "910562452", "88", "88911", "2", 6,
            new BigDecimal("87.353898"), 35, 45, new BigDecimal("257.287398"),
            new BigDecimal("2149.569700")),

        new SykefraværsstatistikkVirksomhetMedGradering(sisteKvartal.getÅrstall(),
            sisteKvartal.getKvartal(), "311874411", "87", "87102", "2", 7,
            new BigDecimal("44.954576"), 21, 59, new BigDecimal("279.808876"),
            new BigDecimal("2049.160900")),

        new SykefraværsstatistikkVirksomhetMedGradering(sisteKvartal.getÅrstall(),
            sisteKvartal.getKvartal(), "315829062", "87", "87102", "2", 0,
            new BigDecimal("0.000000"), 9, 49, new BigDecimal("69.640600"),
            new BigDecimal("1333.733200")),

        new SykefraværsstatistikkVirksomhetMedGradering(sisteKvartal.getÅrstall(),
            sisteKvartal.getKvartal(), "311545795", "88", "88911", "2", 5,
            new BigDecimal("83.618182"), 10, 43, new BigDecimal("121.018182"),
            new BigDecimal("1831.816000")),

        new SykefraværsstatistikkVirksomhetMedGradering(sisteKvartal.getÅrstall(),
            sisteKvartal.getKvartal(), "312679671", "88", "88998", "2", 2,
            new BigDecimal("70.500000"), 11, 59, new BigDecimal("158.800000"),
            new BigDecimal("3208.506000")),

        new SykefraværsstatistikkVirksomhetMedGradering(sisteKvartal.getÅrstall(),
            sisteKvartal.getKvartal(), "910562436", "06", "06100", "2",
            13,
            new BigDecimal("243.251638"), 40, 469, new BigDecimal("809.251638"),
            new BigDecimal("26694.646600"))
    );
  }
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/importAvSykefraværsstatistikk/SykefraværsstatistikkImporteringUtils.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.importAvSykefraværsstatistikk

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.aggregertOgKvartalsvisSykefraværsstatistikk.domene.Varighetskategori
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.aggregertOgKvartalsvisSykefraværsstatistikk.domene.Varighetskategori.TOTAL
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.fellesdomene.SykefraværsstatistikkVirksomhet
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.fellesdomene.SykefraværsstatistikkVirksomhetMedGradering
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.fellesdomene.ÅrstallOgKvartal
import java.math.BigDecimal
import java.math.BigDecimal.ZERO

object SykefraværsstatistikkImporteringUtils {
    fun genererSykefraværsstatistikkVirksomhet(
        gjeldendeKvartal: ÅrstallOgKvartal
    ): List<SykefraværsstatistikkVirksomhet> {
        return HardkodetKildeTilVirksomhetsdata.hentVirksomheter(gjeldendeKvartal).flatMap {
            val muligeDagsverk = (0..100_00).random()
            val tapteDagsverkØvreGrense = muligeDagsverk / 6
            val renTilfeldighet = it.orgnr.verdi.last() == '9'

            Varighetskategori.entries.map { varighet ->
                val tapteDagsverk = (0..tapteDagsverkØvreGrense).random()

                SykefraværsstatistikkVirksomhet(
                    årstall = gjeldendeKvartal.årstall,
                    kvartal = gjeldendeKvartal.kvartal,
                    orgnr = it.orgnr.verdi,
                    varighet = varighet.kode,
                    rectype = it.rectype!!,
                    antallPersoner = if (renTilfeldighet) 0 else (0..500).random(),
                    tapteDagsverk = if (varighet != TOTAL) tapteDagsverk.toBigDecimal() else ZERO,
                    muligeDagsverk = if (varighet == TOTAL) muligeDagsverk.toBigDecimal() else ZERO,
                )
            }
        }
    }

    fun genererSykefraværsstatistikkVirksomhetMedGradering(sisteKvartal: ÅrstallOgKvartal): List<SykefraværsstatistikkVirksomhetMedGradering> {
        return listOf(
            SykefraværsstatistikkVirksomhetMedGradering(
                årstall = sisteKvartal.årstall,
                kvartal = sisteKvartal.kvartal,
                orgnr = "910562452",
                næring = "88",
                næringkode = "88911",
                rectype = "2",
                antallGraderteSykemeldinger = 6,
                tapteDagsverkGradertSykemelding = BigDecimal("87.353898"),
                antallSykemeldinger = 35,
                antallPersoner = 45,
                tapteDagsverk = BigDecimal("257.287398"),
                muligeDagsverk = BigDecimal("2149.569700")
            ),
            SykefraværsstatistikkVirksomhetMedGradering(
                sisteKvartal.årstall,
                sisteKvartal.kvartal,
                "311874411",
                "87",
                "87102",
                "2",
                7,
                BigDecimal("44.954576"),
                21,
                59,
                BigDecimal("279.808876"),
                BigDecimal("2049.160900")
            ),
            SykefraværsstatistikkVirksomhetMedGradering(
                sisteKvartal.årstall,
                sisteKvartal.kvartal,
                "315829062",
                "87",
                "87102",
                "2",
                0,
                BigDecimal("0.000000"),
                9,
                49,
                BigDecimal("69.640600"),
                BigDecimal("1333.733200")
            ),
            SykefraværsstatistikkVirksomhetMedGradering(
                sisteKvartal.årstall,
                sisteKvartal.kvartal,
                "311545795",
                "88",
                "88911",
                "2",
                5,
                BigDecimal("83.618182"),
                10,
                43,
                BigDecimal("121.018182"),
                BigDecimal("1831.816000")
            ),
            SykefraværsstatistikkVirksomhetMedGradering(
                sisteKvartal.årstall,
                sisteKvartal.kvartal,
                "312679671",
                "88",
                "88998",
                "2",
                2,
                BigDecimal("70.500000"),
                11,
                59,
                BigDecimal("158.800000"),
                BigDecimal("3208.506000")
            ),
            SykefraværsstatistikkVirksomhetMedGradering(
                sisteKvartal.årstall,
                sisteKvartal.kvartal,
                "910562436",
                "06",
                "06100",
                "2",
                13,
                BigDecimal("243.251638"),
                40,
                469,
                BigDecimal("809.251638"),
                BigDecimal("26694.646600")
            )
        )
    }
}
